python/cuml/common/array_sparse.py

```diff
@@ -149,7 +149,9 @@ def to_output(self, output_type='cupy',
             if output_dtype is None else output_dtype
 
         if output_type not in ['cupy', 'scipy']:
-            raise ValueError("Unsupported output_type: %s" % output_dtype)
+            # raise ValueError("Unsupported output_type: %s" % output_type)
```

Comment:
This is a special modification I made. It is necessary when you want to return a sparse `SparseCumlArray` despite a dense input.

For instance, when `KBinsDiscretizer` is used with `encode=onehot`, it uses the `OneHotEncoder` model and parameterize it to output a sparse array. In this specific case, you have a dense input but want to output a sparse CuPy array.

In this code `output_type` can takes values such as `dataframe`(pandas), `cudf`, or `numba`. This is questionable, but I thought that instead of raising an exception, we would force the output to be a CuPy sparse array. Otherwise, an estimator that took a dense input would never be able to return a sparse output.